src/ensembl/src/content/app/browser/zmenu/ZmenuInstantDownload.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+import useApiService from 'src/shared/hooks/useApiService';
+
+import { InstantDownloadTranscript } from 'src/shared/components/instant-download';
+import { CircleLoader } from 'src/shared/components/loader/Loader';
+import { TranscriptInResponse } from 'src/content/app/entity-viewer/shared/rest/rest-data-fetchers/transcriptData';
+
+import { LoadingState } from 'src/shared/types/loading-state';
+
+import styles from './Zmenu.scss';
+
+type Props = {
+  id: string;
+};
+
+const ZmenuInstantDownload = (props: Props) => {
+  const transcriptId = getStableId(props.id);
+  const params = {
+    endpoint: `/lookup/id/${transcriptId}?content-type=application/json;expand=1`,
+    host: 'https://rest.ensembl.org'
+  };
+  const { loadingState, data, error } = useApiService<TranscriptInResponse>(
+    params
+  );
+
+  if (loadingState === LoadingState.LOADING) {
+    return (
+      <div className={styles.zmenuInstantDowloadLoading}>
+        <CircleLoader />
+      </div>
+    );
+  }
+
+  if (error) {
```

Comment:
Should there be a todo comment here to revisit this later?

Reply:
🤷‍♂️ but ✔️